cluster/kubernetes/update.go

```diff
@@ -169,14 +175,16 @@ func tryUpdate(def []byte, container string, newImage image.Ref, out io.Writer)
 		return replaceRCNameRE.ReplaceAllString(found, fmt.Sprintf(`${1}%s${2}`, newDefName))
 	})
 
+	// TODO: delete all regular expressions which are used to modify YAML.
+	// See #1019. Modifying this is not recommended.
 	// Replacing labels: these are in two places, the container template and the selector
 	// TODO: This doesn't handle # comments
 	// TODO: This encodes an expectation of map keys being ordered (i.e. name *then* version)
 	// TODO: This assumes that these are indented by exactly 2 spaces (which may not be true)
 	replaceLabelsRE := multilineRE(
 		`((?:  selector|      labels):.*)`,
 		`((?:  ){2,4}name:.*)`,
-		`((?:  ){2,4}version:\s*) (?:` + optq + `[-\w]+` + optq + `)(\s.*)`,
+		`((?:  ){2,4}version:\s*) (?:`+optq+`[-\w]+`+optq+`)(\s.*)`,
```

Comment:
grr who wasn't running gofmt (possibly me)

Reply:
Checking `git blame` might surprise you... or not since you're aware of why we don't change this code.